modules/core/src/smithy4s/http/CodecAPI.scala

```diff
@@ -31,6 +31,9 @@ import internals.StringAndBlobCodecSchemaVisitor
 trait CodecAPI {
 
   type Codec[A]
+  type Cache
+
+  def createCache(): Cache
```

Comment:
Right, so this "hides" the nature of the cache to CodecAPI callers. This is because the Cache might actually be caching constructs that are not for public consumption, so we're relying on a type-member to hide the detail, and passing it around. 

The reason we're not just making a `CachedCodecAPI` implementation is that doing so would be harder to respect binary compatibility, as the "end of the world" currently relies on a `val CodecAPI` being present on the service builders. Because the ServiceBuilder are Global, this wouldn't play too well without breaking bincompat